monai/utils/misc.py

```diff
@@ -322,9 +330,85 @@ def dtype_torch_to_numpy(dtype):
 
 def dtype_numpy_to_torch(dtype):
     """Convert a numpy dtype to its torch equivalent."""
+    # np dtypes can be given as np.float32 and np.dtype(np.float32) so unify them
+    dtype = np.dtype(dtype) if type(dtype) == type else dtype
     return _np_to_torch_dtype[dtype]
 
 
+def dtype_convert(dtype, data_type):
+    """Convert to the `dtype` that corresponds to `data_type`.
+    Example:
+        im = torch.tensor(1)
+        dtype = dtype_convert(np.float32, type(im))
+    """
+    if data_type is torch.Tensor:
+        if type(dtype) is torch.dtype:
+            return dtype
+        return dtype_numpy_to_torch(dtype)
+    else:
+        if type(dtype) is not torch.dtype:
+            return dtype
+        return dtype_torch_to_numpy(dtype)
+
+
+def convert_data_type(
+    data: Any,
+    output_type: Optional[type] = None,
+    device: Optional[torch.device] = None,
```

Comment:
Not sure I fully understand your first point. If `device` is `None`, the object will remain on the same device.

For the second point, I've allowed `output_type` to be `None` so that this function can be used to change the `dtype` of an image without changing numpy/torch, e.g.: `convert_data_type(data, dtype=torch.float32)`.

Reply:
I see, for the 2nd point the current impl. looks good, we can enhance the docstring later...

for the 1st point, I mainly saw this type of input `device=d[key].device if isinstance(d[key], torch.Tensor) else None` a few times, and think perhaps it could be the default behaviour.